File: Sources/VihChatBotSDK/Models/LoanApprovalModels.swift
import Foundation

public struct LoanApprovalRequest: Codable {
    public var session_id: String
    public init(session_id: String) { self.session_id = session_id }
}

/// Response from POST /main/loan-approval/. Top-level envelope carries
/// status/message/errorCode plus flat `callId` and `voiceWsUrl`.
public struct LoanApprovalResponse: Codable {
    public var status: Bool?
    public var message: String?
    public var data: LoanApprovalData?
    public var callId: String?
    public var voiceWsUrl: String?
    public var errorCode: String?

    enum CodingKeys: String, CodingKey {
        case status, message, data
        case callId = "call_id"
        case voiceWsUrl = "voice_ws_url"
        case errorCode = "error_code"
    }
}

public struct LoanApprovalData: Codable {
    public var url: String?
    public var callId: String?
    public var requestId: String?
    public var voicebotId: String?
    public var customerPhone: String?
    public var callSid: String?
    public var status: String?
    public var correlationMethod: String?
    public var timestamp: String?
}

public struct CallDetailsRequest: Codable {
    public var call_id: String
    public var session_id: String
    public init(call_id: String, session_id: String) {
        self.call_id = call_id
        self.session_id = session_id
    }
}

File: vihChatBot/src/main/java/com/vihmessenger/vihchatbot/data/model/LoanApprovalModels.kt
package com.vihmessenger.vihchatbot.data.model

import com.google.gson.annotations.SerializedName

data class LoanApprovalRequest(
    @SerializedName("session_id") val session_id: String
)

/**
 * Response shape from POST /main/loan-approval/.
 *
 * The top-level envelope carries [status]/[message]/[errorCode] plus flat
 * conveniences [callId] and [voiceWsUrl]. The nested [data] block holds the
 * full Ultravox payload (the source of truth for the call). We prefer the
 * flat top-level fields when reading.
 */
data class LoanApprovalResponse(
    @SerializedName("status") val status: Boolean?,
    @SerializedName("message") val message: String?,
    @SerializedName("data") val data: LoanApprovalData?,
    @SerializedName("call_id") val callId: String?,
    @SerializedName("voice_ws_url") val voiceWsUrl: String?,
    @SerializedName("error_code") val errorCode: String?
)

data class LoanApprovalData(
    @SerializedName("url") val url: String?,
    @SerializedName("callId") val callId: String?,
    @SerializedName("requestId") val requestId: String?,
    @SerializedName("voicebot_id") val voicebotId: String?,
    @SerializedName("customerPhone") val customerPhone: String?,
    @SerializedName("callSid") val callSid: String?,
    @SerializedName("status") val status: String?,
    @SerializedName("correlationMethod") val correlationMethod: String?,
    @SerializedName("timestamp") val timestamp: String?
)

data class CallDetailsRequest(
    @SerializedName("call_id") val call_id: String,
    @SerializedName("session_id") val session_id: String
)
